### ISSUE TITLE
Running from javascript/typescript

### ISSUE THREAD

**Issue by ChuckJonas** [microsoft/genaiscript#917]
I have a use case where I would like to process a folder of files and use the output of a genai-script to write rows in a DB. 

Pseudo code would look something like this:
```ts
const previouslyProcess = getProcessedHashes();
const folder = getFolder();
for(file of folder){
   const hash = md5(file)
   if(previouslyProcess.contains(hash)) 
       continue;
   const output = /npx --yes genaiscript@^1.80.10 run my_script file
   writeResults(output)
}
```

**Is the only way to do this to spawn a process and parse the output for stdout?**   Or can it be invoked directly via a javascript api?

_Seems like this would be a very common usecase, so sorry if this is in the docs.  Searched and couldn't find it._

**Comment by ChuckJonas** [comment]
related #450

**Comment by ChuckJonas** [comment]
This is my current workaround... Feels fragile and inefficient, especially if you are going to process a large number of files.  Seems like there is a decent bit of overhead to spin up `genaiscript` each time.

```
const { exec } = require("child_process");
const fs = require("fs");

async function runGenAi(script: string, file: string) {
  return new Promise((resolve, reject) => {
    exec(
      `npx --yes --quiet genaiscript@^1.80.10 run ${script} "${file}" -j`,
      async (error, stdout, stderr) => {
        if (error) {
          console.error(`Error executing command:`, error);
          reject(error);
        } else {
          // trim to first and last {} as there was noise on the stdout
          const output = stdout.trim().match(/\{[\s\S]*\}/);
          try {
            const json = JSON.parse(output);
            if (json["json"]) {
              resolve(json["json"]);
            }
            reject("No JSON found in output");
          } catch (e) {
            console.error(`Error parsing JSON:`, e);
            reject(e);
          }
        }
      }
    );
  });
}

runGenAi("receipts", "purchase.pdf")
  .then(console.log)
  .catch(console.error);
```

**Comment by pelikhan** [comment]
I will export a "run" api

**Comment by pelikhan** [comment]
See https://github.com/microsoft/genaiscript/pull/925

**Comment by pelikhan** [comment]
Available for testing in 1.83.0 https://microsoft.github.io/genaiscript/reference/cli/api/
